pydnd/equipment/views.py

```diff
@@ -1,18 +1,75 @@
 from django.shortcuts import render,get_object_or_404
-
+from rest_framework import generics, status
 # Create your views here.
 from rest_framework.views import APIView
 from rest_framework.response import Response
-from .models import Equipment, Armor, Weapon
-from .serializers import EquipmentListSerializer, ArmorListSerializer, WeaponListSerializer
+from django.forms.models import model_to_dict
+from django.core.exceptions import ObjectDoesNotExist
+from .models import Equipment, Armor, Weapon, EquipmentSubCategory, EquipmentCategory
+from .serializers import EquipmentListSerializer, ArmorListSerializer, WeaponListSerializer,EquipmentSerializer, EquipmentSubCategorySerializer, EquipmentCategorySerializer
 
 
 class EquipmentList(APIView):
-    def get(self, request):
+   def get(self, request):
         equipment = Equipment.objects.all()
         data = EquipmentListSerializer(equipment, many=True).data
         return Response(data)
 
+
+
+class EquipmentCategoryList(generics.ListCreateAPIView):
+
+    queryset = EquipmentCategory.objects.all()
+    serializer_class =  EquipmentCategorySerializer
+
+
+class EquipmentCategoryGet(APIView):
+
+    serializer_class = EquipmentCategorySerializer
+
+    def get(self, request, name_or_id):
+        return get_equipment_category(name_or_id)
```

Comment:
This is probably specific enough to not be it's own method and just live here in this GET.